### PR TITLE
Nine-point circle

### PR DESCRIPTION
**Pull request by enedil**
https://en.wikipedia.org/wiki/Nine-point_circle
Create nine-point circle

### CODE REVIEW

**Comment by smichr** [comment]
+1

File: sympy/geometry/polygon.py [hunk 1]
```diff
@@ -2249,6 +2250,40 @@ def medial(self):
...
+        >>> p1, p2, p3 = Point(0, 0), Point(1, 0), Point(0, 1)
+        >>> t = Triangle(p1, p2, p3)
+        >>> t.nine_point_circle
+        Circle(Point(1/4, 1/4), sqrt(2)/4)
+
+        """
+        return Circle(*self.medial.vertices)
+
+
+
```
Comment (leosartaj) [inline sympy/geometry/polygon.py]: You can remove two extra empty lines here.

**Comment by leosartaj** [comment]
You should add few tests for the new addition.

**Comment by leosartaj** [comment]
Other than my few comments. This looks good. +1

**Comment by enedil** [comment]
Ok, apparently fixed those.
Are you interested in implementation of other triangle centers?

File: sympy/geometry/polygon.py [hunk 2]
```diff
@@ -2249,6 +2250,38 @@ def medial(self):
...
+        sympy.geometry.polygon.Triangle.orthocenter
+
+        Examples
+        ========
+
+        >>> from sympy.geometry import Point, Triangle
+        >>> p1, p2, p3 = Point(0, 0), Point(1, 0), Point(0, 1)
+        >>> t = Triangle(p1, p2, p3)
+        >>> t.nine_point_circle
+        Circle(Point(1/4, 1/4), sqrt(2)/4)
```
Comment (leosartaj) [inline sympy/geometry/polygon.py]: ``Circle(Point2D(1/4, 1/4), sqrt(2)/4)``

File: sympy/geometry/tests/test_polygon.py [hunk 3]
```diff
@@ -228,6 +228,11 @@ def test_polygon():
     assert intersection(m[p1], m[p2], m[p3]) == [t1.centroid]
     assert t1.medial == Triangle(Point(2.5, 0), Point(0, 2.5), Point(2.5, 2.5))
 
+    # Nine-point circle
+    vo_middles = [Segment(v, t1.orthocenter).midpoint for v in t1.vertices]
```
Comment (leosartaj) [inline sympy/geometry/tests/test_polygon.py]: ``vo_middles`` in this case is ``[Point2D(2, 0), <bound method Point2D.midpoint of Point2D(4, 0)>, Point2D(4, 3/2)]``, probably due to t1 being a right-angle triangle.

Comment (enedil) [inline sympy/geometry/tests/test_polygon.py]: @leosartaj Yeah, since `t1` is both right and isosceles, 9 points are degenerated to only 4 distinct.

Comment (leosartaj) [inline sympy/geometry/tests/test_polygon.py]: You have to probably change this line, as this causes test failure.

Comment (enedil) [inline sympy/geometry/tests/test_polygon.py]: Yeah, I even did commit, albeit did not pull.

**Comment by leosartaj** [comment]
This looks good. Merging.

**Comment by leosartaj** [comment]
Thanks @enedil !

**Pull request merged**